packages/cdktf-cli/bin/cmds/helper/check-environment.ts

```diff
@@ -64,3 +65,52 @@ export async function checkEnvironment(projectPath = process.cwd()) {
       await checkGoVersion();
   }
 }
+
+export async function verifySimilarLibraryVersion() {
+  let output;
+  try {
+    output = await exec("npm", ["list", "cdktf", "--json"], {
+      env: process.env,
+    });
```

Comment:
This only works in TS projects, doesn't it?

Reply:
🤦 yeah good point, I sort of forgot that we install cdktf through different package managers. I'll restrict the check to TS

Reply:
I changed it so we can add support for more languages down the road 👍